components/job-orchestration/job_orchestration/scheduler/compress/compression_scheduler.py

```diff
@@ -518,7 +520,16 @@ def main(argv):
     logger.info(f"Starting {COMPRESSION_SCHEDULER_COMPONENT_NAME}")
     sql_adapter = SqlAdapter(clp_config.database)
 
-    task_manager = CeleryTaskManager()
+    if clp_config.compression_scheduler.type == OrchestrationType.celery:
```

Comment:
```suggestion
    task_manager: CeleryTaskManager | SpiderTaskManager
    if clp_config.compression_scheduler.type == OrchestrationType.celery:
```
Even in Python what u wrote should work, I'd still suggest to forward declare the variable so that the scope is more clear.